### PR TITLE
[No Jira] Check if StateListAnimator is supported

### PR DESCRIPTION
**Pull request by pietmau**
Need to check that api is at least 22 (LOLLIPOP_MR1) ☹️

### CODE REVIEW

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package net.skyscanner.backpack.util
+
+import android.animation.AnimatorInflater
+import android.os.Build
+import android.view.View
+import androidx.annotation.DrawableRes
+
+fun View.isStateListAnimatorSupported() =
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: I don't think we need to make it public.

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP) {
+    false
+  } else isSpecificDeviceSupported()
+
+private fun isSpecificDeviceSupported() = Build.MODEL != "GT-I9505"
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: We need to check manufacturer as well

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+fun View.loadStateListAnimator(@DrawableRes animator: Int) {
+  stateListAnimator = AnimatorInflater.loadStateListAnimator(context, animator)
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: It's not clear what stateListAnimator is. Please add this prefix.

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import androidx.annotation.DrawableRes
+
+fun View.isStateListAnimatorSupported() =
+  if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP) {
+    false
+  } else isSpecificDeviceSupported()
+
+private fun isSpecificDeviceSupported() = Build.MODEL != "GT-I9505"
+
+fun View.loadStateListAnimator(@DrawableRes animator: Int) {
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: Also internal. Plus it's worth consideration to move all these methods to BpkButton

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt (hunk 1 above)
Comment (pietmau) [reply Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: I agree, internal is better

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt (hunk 3 above)
Comment (pietmau) [reply Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: yeah

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt (hunk 4 above)
Comment (pietmau) [reply Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: yup

File: Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt (hunk 2 above)
Comment (pietmau) [reply Backpack/src/main/java/net/skyscanner/backpack/util/extensions.kt]: as discussed

**Comment by tiagoengel** [comment]
closed in favour of #285

**Pull request closed without merging**